### PR TITLE
De-virtualize toBasetype() so it is inlineable

### PR DESCRIPTION
**Pull request by WalterBright**
@andralex profiled it as being in the dirty dozen timewasters in dmd. Address that by de-virtualizing it and making it inlineable.

### CODE REVIEW

**Comment by NilsLankila** [comment]
is the list of the dozen dirty timewasters published somewhere ?

**Review by thewilsonator — APPROVED** [review]

File: src/dmd/mtype.d [hunk 1]
```diff
@@ -5966,7 +5970,7 @@ extern (C++) final class TypeEnum : Type
         return MATCH.nomatch;
     }
 
-    override Type toBasetype()
+    final Type toBasetype2()
```
Comment (ibuclaw) [inline src/dmd/mtype.d:5973]: Maybe toMemtype?  I'm not bothered either way.  Also `extern(D)` so that it is not exported to C++ please.

Comment (ibuclaw) [reply src/dmd/mtype.d:5973]: As this is now a unique function, it would also be good to add some Ddoc about what it does.

Comment (WalterBright) [reply src/dmd/mtype.d:5974]: I did the latter.

**Comment by UplinkCoder** [comment]
@WalterBright have you confirmed, via profiling, that this actually helps?

**Pull request merged**